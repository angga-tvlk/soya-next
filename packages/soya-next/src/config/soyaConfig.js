import config from 'config';

const stringifyConfig = config => (function recursive(config, stringified, prevKeys) {
  return Object.keys(config).reduce((stringified, key) => {
    const allKeys = prevKeys.concat(key);
    const allKey = `soya.config.${allKeys.join('.')}`;
    if (typeof config[key] === 'object') {
      stringified[allKey] = config[key];
      return recursive(config[key], stringified, allKeys);
    }
    stringified[allKey] = config[key];
    return stringified;
  }, stringified);
})(config, {}, []);

export const raw = () => ({
  soya: {
    config: Object.keys(config).reduce((newConfig, key) => {
      newConfig[key] = config[key];
      return newConfig;
    }, {}),
  },
});

export const stringify = () => stringifyConfig(config);
