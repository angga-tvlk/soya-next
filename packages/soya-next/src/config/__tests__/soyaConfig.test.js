import { stringify, raw } from '../soyaConfig';

describe('Soya Config', () => {
  it('should return raw config object', () => {
    expect(raw()).toMatchSnapshot();
  });

  it('should return stringify config', () => {
    expect(stringify()).toMatchSnapshot();
  });
});
