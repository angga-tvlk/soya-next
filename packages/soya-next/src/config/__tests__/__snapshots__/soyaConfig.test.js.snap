// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Soya Config should return raw config object 1`] = `
Object {
  "soya": Object {
    "config": Object {
      "deepArray": Array [
        Array [
          "item 1.1",
          "item 1.2",
        ],
        Array [
          "item 2.1",
          "item 2.2",
        ],
      ],
      "deepObject": Object {
        "shallowObject": Object {
          "key": "value",
        },
      },
      "key": 123,
      "shallowArray": Array [
        "item 1",
        "item 2",
      ],
      "shallowObject": Object {
        "key": "value",
      },
    },
  },
}
`;

exports[`Soya Config should return stringify config 1`] = `
Object {
  "soya.config.deepArray": Array [
    Array [
      "item 1.1",
      "item 1.2",
    ],
    Array [
      "item 2.1",
      "item 2.2",
    ],
  ],
  "soya.config.deepArray.0": Array [
    "item 1.1",
    "item 1.2",
  ],
  "soya.config.deepArray.0.0": "item 1.1",
  "soya.config.deepArray.0.1": "item 1.2",
  "soya.config.deepArray.1": Array [
    "item 2.1",
    "item 2.2",
  ],
  "soya.config.deepArray.1.0": "item 2.1",
  "soya.config.deepArray.1.1": "item 2.2",
  "soya.config.deepObject": Object {
    "shallowObject": Object {
      "key": "value",
    },
  },
  "soya.config.deepObject.shallowObject": Object {
    "key": "value",
  },
  "soya.config.deepObject.shallowObject.key": "value",
  "soya.config.key": 123,
  "soya.config.shallowArray": Array [
    "item 1",
    "item 2",
  ],
  "soya.config.shallowArray.0": "item 1",
  "soya.config.shallowArray.1": "item 2",
  "soya.config.shallowObject": Object {
    "key": "value",
  },
  "soya.config.shallowObject.key": "value",
}
`;
